client/src/containers/MultiFactorApp.js

```diff
@@ -11,20 +12,27 @@ class MultiFactorApp extends Component {
       schema: null,
       schemaLoaded: false,
     };
+
+    this.handleCompleteLogin = this.handleCompleteLogin.bind(this);
   }
 
   componentDidMount() {
-    const thisComponent = this;
     const { schemaURL } = this.props;
     return fetch(schemaURL)
       .then(response => response.json())
       .then(schemaData =>
-        thisComponent.setState({
+        this.setState({
           schema: schemaData
         })
       );
   }
 
+  handleCompleteLogin() {
+    const { schema: { endpoints: { complete } } } = this.state;
+
+    window.location = complete;
```

Comment:
Seems better than requiring react-router just for this 👍

Reply:
Yep 😆 . I did actually briefly consider some sort of library for this.